testing/integration/endpoint_security_test.go

```diff
@@ -87,54 +89,129 @@ func TestEndpointSecurity(t *testing.T) {
+			kibana.MonitoringEnabledMetrics,
+		},
+	}
+	policyResp, err := tools.InstallAgentWithPolicy(t, fixture, info.KibanaClient, createPolicyReq)
+
+	t.Log("Installing Elastic Defend")
+	installElasticDefendPackage(t, info, policyResp.ID)
+
+	t.Log("Polling for endpoint-security to become Healthy")
+	ctx, cancel := context.WithTimeout(context.Background(), endpointHealthPollingTimeout)
+	defer cancel()
+
+	agentClient := fixture.Client()
+	err = agentClient.Connect(ctx)
+	require.NoError(t, err)
+
+	require.Eventually(t,
+		func() bool { return endpointIsHealthy(t, ctx, agentClient) },
+		endpointHealthPollingTimeout,
+		time.Second,
+		"Endpoint component or units are not healthy.",
+	)
+	t.Logf("Verified endpoint component and units are healthy")
 
-		foundEndpointInputUnit := false
-		foundEndpointOutputUnit := false
-		for _, comp := range state.Components {
-			isEndpointComponent := strings.Contains(comp.Name, "endpoint")
-			if comp.State != client.Healthy {
-				t.Logf("Component is not Healthy\n%+v", comp)
+	// Unenroll the agent
+	t.Logf("Unenrolling the agent")
+
+	hostname, err := os.Hostname()
+	require.NoError(t, err)
+
+	agentID, err := tools.GetAgentIDByHostname(info.KibanaClient, hostname)
+	require.NoError(t, err)
+
+	_, err = info.KibanaClient.UnEnrollAgent(kibana.UnEnrollAgentRequest{ID: agentID})
+	require.NoError(t, err)
+
+	t.Logf("Waiting for inputs to stop")
+	require.Eventually(t,
+		func() bool {
+			state, err := agentClient.State(ctx)
+			if err != nil {
+				t.Logf("Error getting agent state: %s", err)
 				return false
 			}
 
-			for _, unit := range comp.Units {
-				if isEndpointComponent {
-					if unit.UnitType == client.UnitTypeInput {
-						foundEndpointInputUnit = true
-					}
-					if unit.UnitType == client.UnitTypeOutput {
-						foundEndpointOutputUnit = true
-					}
-				}
+			if state.State != client.Healthy {
+				t.Logf("Agent is not Healthy\n%+v", state)
+				return false
+			}
 
-				if unit.State != client.Healthy {
-					t.Logf("Unit is not Healthy\n%+v", unit)
-					return false
-				}
+			if len(state.Components) != 0 {
+				t.Logf("Components have not been stopped and uninstalled!\n%+v", state)
+				return false
 			}
-		}
 
-		// Ensure both the endpoint input and output units were found and healthy.
-		if !foundEndpointInputUnit || !foundEndpointOutputUnit {
-			t.Logf("State did not contain endpoint units!\n%+v", state)
-			return false
+			if state.FleetState != client.Failed {
+				t.Logf("Fleet state has not been marked as failed yet!\n%+v", state)
+				return false
+			}
+
+			return true
+		},
+		endpointHealthPollingTimeout,
+		time.Second,
+		"All components not removed.",
+	)
+	t.Logf("Verified endpoint component and units are removed")
+
+	// Verify that the Endpoint directory was correctly removed.
+	// Regression test for https://github.com/elastic/elastic-agent/issues/3077
+	agentInstallPath := fixture.WorkDir()
+	files, err := os.ReadDir(filepath.Join(agentInstallPath, ".."))
```

Comment:
Nitpick: since we are going 1 level up from the agent install maybe we want to [clean the path](https://pkg.go.dev/path/filepath#Clean) using `filepath.Clean(filepath.Join(...))` 

We could also use [`filepath.SplitList()`](https://pkg.go.dev/path/filepath#SplitList) + `filepath.Join` removing the last element
I think also `filepath.Dir()` should work here (to be tested)